From 363497d092255914781f951376e9090f67bb9d13 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:40 -0400
Subject: [fs] nfsd: read size estimate should include padding

Message-id: <1403560314-32525-99-git-send-email-bfields@redhat.com>
Patchwork-id: 84546
O-Subject: [Patch RHEL7 098/172] nfsd4: read size estimate should include padding
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 622f560e6afbd54cdd9b9a416ddab8114baedd76
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4proc.c b/fs/nfsd/nfs4proc.c
index c2b7593..31ce60f 100644
--- a/fs/nfsd/nfs4proc.c
+++ b/fs/nfsd/nfs4proc.c
@@ -1488,7 +1488,7 @@ static inline u32 nfsd4_read_rsize(struct svc_rqst *rqstp, struct nfsd4_op *op)
  if (rlen > maxcount)
   rlen = maxcount;
 
- return (op_encode_hdr_size + 2) * sizeof(__be32) + rlen;
+ return (op_encode_hdr_size + 2 + XDR_QUADLEN(rlen)) * sizeof(__be32);
 }
 
 static inline u32 nfsd4_readdir_rsize(struct svc_rqst *rqstp, struct nfsd4_op *op)
-- 
1.7.1